### PR TITLE
[WT-317] feature/Added profile photo to cache

### PR DESCRIPTION
**Pull request by CandelR**
- Added profile avatar image to cache

### CODE REVIEW

File: src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      const handleAvatarData = async () => {
+        if (avatarSrcURL) {
+          await handleDownload(avatarSrcURL);
+          return;
+        }
+        deleteDatabaseProfileAvatar();
+        setAvatarBlob(null);
+      };
+
+      handleAvatarData();
```
Comment (sg-gs) [inline src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx:55]: Handle the error here please

File: src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import { memo, useEffect, useState } from 'react';
+import { deleteDatabaseProfileAvatar, getDatabaseProfileAvatar } from '../../../../../drive/services/database.service';
+import Avatar from '../../../../../shared/components/Avatar';
+import { useAppDispatch } from '../../../../../store/hooks';
+import { updateUserAvatarThunk } from '../../../../../store/slices/user';
+
+export const extractExpiresValue = (url: string): number | null => {
```
Comment (sg-gs) [inline src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx]: Remove this function

File: src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  const regex = /Expires=(\d+)/;
+  const match = url.match(regex);
+  return match ? parseInt(match[1]) : null;
+};
+
+export const extractAvatarURLID = (url: string): string | null => {
```
Comment (sg-gs) [inline src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx:9]: Remove this function if not required

**Review by sg-gs — CHANGES_REQUESTED** [review]

File: src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx (hunk 3 above)
Comment (CandelR) [reply src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx:9]: is used to extract the uuid from the avatar url and compare it with the uuid in the cache.

Comment (sg-gs) [reply src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx:9]: But, the avatar UUID is in `xUser.avatar`, or at least, it should be

Comment (sg-gs) [reply src/app/core/views/Preferences/tabs/Account/AvatarWrapper.tsx:9]: Ah, okay, the xUser.avatar has the presigned-url, not just the UUID, okay then

**Review by sg-gs — APPROVED** [review]

**Pull request merged**